src/zenml/steps/utils.py

```diff
@@ -403,3 +408,67 @@ def f3(condition):
     visitor.visit(tree)
 
     return visitor.has_only_none_returns
+
+
+def log_step_metadata(
+    metadata: Dict[str, "MetadataType"],
+    step_name: Optional[str] = None,
+    pipeline_name_id_or_prefix: Optional[Union[str, UUID]] = None,
+    pipeline_version: Optional[str] = None,
+    run_id: Optional[str] = None,
+) -> None:
+    """Logs step metadata.
+
+    Args:
+        metadata: The metadata to log.
+        step_name: The name of the step to log metadata for. Can be omitted
+            when being called inside a step.
+        pipeline_name_id_or_prefix: The name of the pipeline to log metadata
+            for. Can be omitted when being called inside a step.
+        pipeline_version: The version of the pipeline to log metadata for.
+            Can be omitted when being called inside a step.
+        run_id: The ID of the run to log metadata for. Can be omitted when
+            being called inside a step.
+
+    Raises:
+        ValueError: If no step name is provided and the function is not called
+            from within a step or if no pipeline name or ID is provided and
+            the function is not called from within a step.
+    """
+    if not step_name:
+        try:
+            step_context = get_step_context()
+            step_name = step_context.step_name
+        except RuntimeError:
+            step_context = None
+    else:
+        step_context = None
```

Comment:
Not critical, but you can do `step_context = None` before `if not step_name:` and do `pass` on the exception catch. Will be the same logic, but 1 line less 🙂

Reply:
I shaved 2 lines off ;)

Reply:
👑